Ladybird/CMakeLists.txt

```diff
@@ -235,13 +235,16 @@ add_dependencies(ladybird ${ladybird_helper_processes})
 function(create_ladybird_bundle target_name)
     set_target_properties(${target_name} PROPERTIES
         OUTPUT_NAME "Ladybird"
-        MACOSX_BUNDLE_GUI_IDENTIFIER org.SerenityOS.Ladybird
-        MACOSX_BUNDLE_BUNDLE_VERSION ${PROJECT_VERSION}
-        MACOSX_BUNDLE_SHORT_VERSION_STRING ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
-        MACOSX_BUNDLE_INFO_PLIST "${SERENITY_SOURCE_DIR}/Ladybird/Info.plist"
         MACOSX_BUNDLE TRUE
         WIN32_EXECUTABLE TRUE
-        XCODE_ATTRIBUTE_PRODUCT_BUNDLE_IDENTIFIER org.SerenityOS.Ladybird
+        XCODE_ATTRIBUTE_PRODUCT_BUNDLE_IDENTIFIER org.SerenityOS.${target_name}
+        MACOSX_BUNDLE_INFO_PLIST ${SERENITY_SOURCE_DIR}/Ladybird/Info.plist.in
+        MACOSX_BUNDLE_GUI_IDENTIFIER org.SerenityOS.${target_name}
+        MACOSX_BUNDLE_ICON_FILE app_icon.icns
+        MACOSX_BUNDLE_BUNDLE_VERSION ${PROJECT_VERSION}
+        MACOSX_BUNDLE_SHORT_VERSION_STRING ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
+        MACOSX_BUNDLE_INFO_STRING ${PROJECT_DESCRIPTION}
+        MACOSX_BUNDLE_COPYRIGHT "Copyright (c) 2018-2024, the SerenityOS developers"
```

Comment:
Hmm. this is interesting. I do wonder how this will affect the experimental GN build though. Currently for that guy we just copy the Info.plist wholesale. 

<https://github.com/SerenityOS/serenity/blob/a968bc6f809d1b1eec60c3e2dbb1255b0d62d1de/Meta/gn/secondary/Ladybird/BUILD.gn#L313-L316>

@nico is there any sort of builtin plist-generation in gn? or would we have to add a skript/action to replicate this CMake behavior?

Reply:
Looks like you can do the same thing in GN: https://gn.googlesource.com/gn/+/refs/heads/main/examples/ios/app/resources/Info.plist

Reply:
The only difference being GN uses the `$(VAR_NAME)` syntax while cmake is using `${VAR_NAME}`. Not sure if you could potentially just pass the file contents as a string and just do an in-place regex to replace the brackets with parentheses.

Reply:
Wellllllll everything in GN is defined by "you" (for the most part). For example all that merge_plist logic is handled by the files in examples/iOS/build 😁 <https://gn.googlesource.com/gn/+/refs/heads/main/examples/ios/build/config/ios/templates/>

I think, while it's neat that we can generate this data for the ladybird app bundle from CMake, it would be a shorter putt to fill in the missing fields in our info.plist with the ones you've added. Like, hard code them. At least until we have multiple app bundles that we want to share a template info.plist from. 

I suppose we could use the same template for both Ladybird and MacPDF, but that probably also needs some massaging of config files.

Reply:
> Wellllllll everything in GN is defined by "you" (for the most part). For example all that merge_plist logic is handled by the files in examples/iOS/build 😁 [gn.googlesource.com/gn/+/refs/heads/main/examples/ios/build/config/ios/templates](https://gn.googlesource.com/gn/+/refs/heads/main/examples/ios/build/config/ios/templates/)
> 
> I think, while it's neat that we can generate this data for the ladybird app bundle from CMake, it would be a shorter putt to fill in the missing fields in our info.plist with the ones you've added. Like, hard code them. At least until we have multiple app bundles that we want to share a template info.plist from.
> 
> I suppose we could use the same template for both Ladybird and MacPDF, but that probably also needs some massaging of config files.

Absolutely; if you feel like it'd be more maintenance this way we can hardcode it. I did it this way to make the `plist` somewhat generic and sharable between the AppKit version and a potential iOS on macOS hybrid app (the thing I was looking into on discord earlier) and to make sure the plist never gets out of sync with the cmake project definition (which would admittedly only really be a problem if we we're doing something like CI/nightly testflight builds).

Let me know if you'd rather have it hardcoded, a script put together that works on both cmake and gn or if we should go with the cmake solution and do something custom in gn. :)

Reply:
Right, y have to use a script. Chromium uses https://source.chromium.org/chromium/chromium/src/+/main:build/apple/tweak_info_plist.py

This patch here looks very similar to configure_file(), which this here emulates: https://github.com/llvm/llvm-project/blob/main/llvm/utils/gn/build/write_cmake_config.py

(Both have a .gni next to it.)

Personally, I like not generating files if I don't have to. `grep` works, you can just look at things, etc. (assuming it doesn't lead to a lot of duplication; haven't read this in detail to have an opinion of that applies here.)

Reply:
I think for the first go around, let's add the extra options to the Info.plist statically. if you have ideas on how to unify our Info.plists for our existing two apps, or we come up with more apps we want to bundle for macOS/iOS, then revisiting the generation steps for both gn and CMake would be a good idea at that time.

Reply:
> I think for the first go around, let's add the extra options to the Info.plist statically. if you have ideas on how to unify our Info.plists for our existing two apps, or we come up with more apps we want to bundle for macOS/iOS, then revisiting the generation steps for both gn and CMake would be a good idea at that time.

Sounds gut! Closing for now :)